portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +33,102 @@
 public class DataServlet extends HttpServlet {
+  private List<String> quotes;
+  Gson gson = new Gson();
+
+      @Override
+    public void init() {
+    quotes = new ArrayList<>();
+    quotes.add(
+        "A ship in port is safe, but that is not what ships are for. "
+            + "Sail out to sea and do new things. - Grace Hopper");
+    quotes.add("They told me computers could only do arithmetic. - Grace Hopper");
+    quotes.add("It is much easier to apologise than it is to get permission. - Grace Hopper");
+    quotes.add("If you can't give me poetry, can't you give me poetical science - Ada Lovelace");
+    quotes.add("I am in a charming state of confusion. - Ada Lovelace");
+    quotes.add(
+        "The Analytical Engine weaves algebraic patterns, "
+            + "just as the Jacquard loom weaves flowers and leaves. - Ada Lovelace");
+    quotes.add(
+        "Sometimes it is the people no one can imagine anything of "
+            + "who do the things no one can imagine. - Alan Turing");
+    quotes.add("Those who can imagine anything, can create the impossible. - Alan Turing");
+  }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    // Get Random Quote and Print to Home Page
+    String quote = quotes.get((int) (Math.random() * quotes.size()));
+    String json = gson.toJson(quote);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+
+    // Load comments from Datastore and print to Home Page
+    Query query = new Query("Task").addSort("firstname", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<String> tasks = new ArrayList<>();
+    List<Task> listOfTasks = new ArrayList<>();
+
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String firstName = (String) entity.getProperty("firstname");
+      String lastName = (String) entity.getProperty("lastname");
+      String country = (String) entity.getProperty("country");
+      String subject = (String) entity.getProperty("subject");
+
+    Task task = new Task(id, firstName, lastName, country, subject);
+    listOfTasks.add(task);
+
+    String info = subject + " - " + firstName + " " + lastName + ", " + country; 
+    response.getWriter().println(gson.toJson(info));
+    
+
+
+
+     // tasks.add(firstName + " " + lastName + ", from " + country + "\n" + "subject: " + subject);
+    }
+
+    response.setContentType("application/json;");
+    //response.getWriter().println(gson.toJson(tasks));
+
+  }
+
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // Store entities to Datastore
+    String firstName = request.getParameter("firstname");
```

Comment:
For each param key, you could convert them to private static final fields to share between #doPost and #doGet. This would reduce the change of typos/mistakes when copy-pasting them around and make it easier to change the key (in one place instead of everywhere). For the field names, the style should be all caps with underscores (ie. FIRST_NAME).

Reply:
When I did that I got this error:
![image](https://user-images.githubusercontent.com/22799608/84189307-ecc6af80-aa49-11ea-89a1-5caf2b333d3a.png)

And I also couldn't initialize it because it's abstract.

Reply:
I think she meant using the following:

```java
private static final String FIRST_NAME = "firstname";
private static final String LAST_NAME = "lastname";
etc.
```

and then, in your `doPost` function:
```java
String firstName = request.getParameter(FIRST_NAME);
```

Reply:
got it, thanks!